### DISCUSSION TITLE
Does NuGet support adding custom feeds through http(s) (using SAS tokens)

### DISCUSSION THREAD

**Discussion by Niharikadutta** [NuGet/Home#10576] [Q&A]
I had a very basic question about NuGet feeds and whether it is possible to add a custom feed (say Azure Storage account) through NuGet by using the http(s) URL of the storage account (and passing SAS token).
Just trying to compare with [Conda custom channels](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/create-custom-channels.html#creating-custom-channels) and see if there is a parallel for NuGet.

**Comment by loic-sharma** [comment]
From my understanding, NuGet supports the following authenticated feed mechanisms:

1. [HTTP authentication](https://docs.microsoft.com/en-us/nuget/reference/nuget-config-file#package-source-sections)
1. [Client certificate authentication](https://github.com/NuGet/Home/blob/20dabe750cc26162346bafcba138cebe21ae2305/proposed/archive/NuGet-ClientCertificates.md)

Unfortunately, Azure Storage authentication using SAS tokens is not supported. You will need to create a custom authentication plugin. See: https://docs.microsoft.com/en-us/nuget/reference/extensibility/nuget-cross-platform-authentication-plugin

**Reply by zivkan** [reply]
Something I've seen elsewhere, was a NuGet v3 json layout on some endpoint, using NuGet's authentication, but downloads are HTTP redirects to blob storage with a SAS token. This way the service doesn't need to be a reverse proxy to the blob storage.  It's another idea.

**Reply by Niharikadutta** [reply]
Interesting, thanks for that idea! Would you happen to have an example I could look at somewhere?
